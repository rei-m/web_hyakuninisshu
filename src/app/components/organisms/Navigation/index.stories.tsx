import { Meta, StoryObj } from '@storybook/react';
import { Navigation } from './index';

export default {
  title: 'organisms/Navigation',
  component: Navigation,
  decorators: [(story) => <div style={{ width: '500px' }}>{story()}</div>],
} as Meta<typeof Navigation>;

type Story = StoryObj<typeof Navigation>;

export const Default: Story = {
  args: {},
};
