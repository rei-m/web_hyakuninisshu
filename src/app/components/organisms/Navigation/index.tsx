import type { MenuType } from '@/app/components/molecules/NavigationItem';
import type { SxAppProps } from '@/theme';

import Link from 'next/link';
import Box from '@mui/material/Box';
import NavigationItem from '@/app/components/molecules/NavigationItem';

import { WIDTH_SIDE_NAV } from '@/theme';
import { ROUTING } from '@/configs/routing';

export type NavigationProps = {
  currentMenuType?: MenuType;
  sx?: SxAppProps;
};

const LINK_SX: SxAppProps = {
  flexGrow: {
    xs: 1,
    sm: 0,
  },
  width: {
    xs: '100%',
    sm: WIDTH_SIDE_NAV,
  },
};

const Navigation = ({ currentMenuType, sx }: NavigationProps) => (
  <Box
    component={'nav'}
    sx={[
      {
        display: 'flex',
        backgroundColor: 'primary.main',
        boxSizing: 'border-box',
        boxShadow: 1,
      },
      ...(Array.isArray(sx) ? sx : [sx]),
    ]}
  >
    <Box component={Link} href={ROUTING.training()} sx={LINK_SX}>
      <NavigationItem menuType={'training'} active={currentMenuType === 'training'} />
    </Box>
    <Box component={Link} href={ROUTING.exam()} sx={LINK_SX}>
      <NavigationItem menuType={'exam'} active={currentMenuType === 'exam'} />
    </Box>
    <Box component={Link} href={ROUTING.karutas()} sx={LINK_SX}>
      <NavigationItem menuType={'material'} active={currentMenuType === 'material'} />
    </Box>
    <Box component={Link} href={ROUTING.about()} sx={LINK_SX}>
      <NavigationItem menuType={'other'} active={currentMenuType === 'other'} />
    </Box>
  </Box>
);

export default Navigation;
