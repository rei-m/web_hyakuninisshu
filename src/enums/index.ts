// このIFは苦肉の策だが他にいい方法はあるのか。。。
export interface EnumCollection<T> {
  readonly values: T[];
  readonly valueOf: (value: T) => T;
}

export enum MenuType {
  Training,
  Exam,
  Material,
  Other
}

export enum QuestionState {
  InAnswer,
  Answered,
  ConfirmCorrect,
  Finished
}

export enum RangeFromCondition {
  One = 1,
  Eleven = 11,
  TwentyOne = 21,
  ThirtyOne = 31,
  FortyOne = 41,
  FiftyOne = 51,
  SixtyOne = 61,
  SeventyOne = 71,
  EightyOne = 81,
  NinetyOne = 91
}

export const RangeFromConditions: EnumCollection<RangeFromCondition> = {
  valueOf: (value: number) => {
    const result = RangeFromConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown RangeFrom, value is ${value}`);
  },
  values: [
    RangeFromCondition.One,
    RangeFromCondition.Eleven,
    RangeFromCondition.TwentyOne,
    RangeFromCondition.ThirtyOne,
    RangeFromCondition.FortyOne,
    RangeFromCondition.FiftyOne,
    RangeFromCondition.SixtyOne,
    RangeFromCondition.SeventyOne,
    RangeFromCondition.EightyOne,
    RangeFromCondition.NinetyOne
  ]
};

export enum RangeToCondition {
  Ten = 10,
  Twenty = 20,
  Thirty = 30,
  Forty = 40,
  Fifty = 50,
  Sixty = 60,
  Seventy = 70,
  Eighty = 80,
  Ninety = 90,
  OneHundred = 100
}

export const RangeToConditions: EnumCollection<RangeToCondition> = {
  valueOf: (value: number) => {
    const result = RangeToConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown RangeTo, value is ${value}`);
  },
  values: [
    RangeToCondition.Ten,
    RangeToCondition.Twenty,
    RangeToCondition.Thirty,
    RangeToCondition.Forty,
    RangeToCondition.Fifty,
    RangeToCondition.Sixty,
    RangeToCondition.Seventy,
    RangeToCondition.Eighty,
    RangeToCondition.Ninety,
    RangeToCondition.OneHundred
  ]
};

export enum KimarijiCondition {
  None = 0,
  One = 1,
  Two = 2,
  Three = 3,
  Four = 4,
  Five = 5,
  Six = 6
}

export const KimarijiConditions: EnumCollection<KimarijiCondition> = {
  valueOf: (value: number) => {
    const result = KimarijiConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown Kimariji, value is ${value}`);
  },
  values: [
    KimarijiCondition.None,
    KimarijiCondition.One,
    KimarijiCondition.Two,
    KimarijiCondition.Three,
    KimarijiCondition.Four,
    KimarijiCondition.Five,
    KimarijiCondition.Six
  ]
};

export enum ColorCondition {
  None = '',
  Blue = 'blue',
  Pink = 'pink',
  Yellow = 'yellow',
  Green = 'green',
  Orange = 'orange'
}

export const ColorConditions: EnumCollection<ColorCondition> = {
  valueOf: (value: string) => {
    const result = ColorConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown Color, value is ${value}`);
  },
  values: [
    ColorCondition.None,
    ColorCondition.Blue,
    ColorCondition.Pink,
    ColorCondition.Yellow,
    ColorCondition.Green,
    ColorCondition.Orange
  ]
};

export enum KarutaStyleCondition {
  KanjiAndKana = 0,
  KanaOnly = 1
}

export const KarutaStyleConditions: EnumCollection<KarutaStyleCondition> = {
  valueOf: (value: number) => {
    const result = KarutaStyleConditions.values.find(v => v === value);
    if (result !== undefined) {
      return result;
    }
    throw new Error(`unknown KarutaStyle, value is ${value}`);
  },
  values: [KarutaStyleCondition.KanjiAndKana, KarutaStyleCondition.KanaOnly]
};
