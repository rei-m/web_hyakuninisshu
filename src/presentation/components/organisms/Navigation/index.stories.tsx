import React from 'react';
import { storiesOf } from '@storybook/react';
import Navigation from './index';

storiesOf('organisms/Navigation', module).add('default', () => <Navigation />);
