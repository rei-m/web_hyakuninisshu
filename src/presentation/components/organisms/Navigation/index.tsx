import React from 'react';
import { GatsbyLinkProps, Link } from 'gatsby';
import clsx from 'clsx';
import makeStyles from '@material-ui/core/styles/makeStyles';
import { paths } from '@src/presentation/routes';
import NavigationItem, { MenuType } from '@src/presentation/components/molecules/NavigationItem';
import { ExamIcon, MaterialIcon, TrainingIcon, OtherIcon } from '@src/presentation/components/atoms/MenuIcon';
import { withRipple } from '@src/presentation/enhancers/withRipple';
import { ThemeInterface } from '@src/presentation/styles/theme';

export type Props = {
  currentMenuType?: MenuType;
  className?: string;
};

const useStyles = makeStyles<ThemeInterface>(theme => ({
  root: {
    display: 'flex',
    backgroundColor: theme.palette.primary.main,
    boxSizing: 'border-box',
    boxShadow: theme.elevationShadow1x,
  },
  itemWrapper: {
    flexGrow: 1,
    width: '100%',
    [`@media screen and (min-width:${theme.minWidthWide})`]: {
      flexGrow: 0,
      height: theme.sideNavWidth,
    },
  },
}));

const LinkWithRipple = withRipple<GatsbyLinkProps<{}>>(Link);

const ICON_MAP = {
  training: TrainingIcon,
  exam: ExamIcon,
  material: MaterialIcon,
  other: OtherIcon,
} as const;

const renderIcon = (menuType: MenuType) => {
  const Icon = ICON_MAP[menuType];
  return <Icon style={{ fontSize: '2.4rem' }} />;
};

const WrappedItem = ({
  to,
  menuType,
  active,
  className,
}: {
  to: string;
  menuType: MenuType;
  active: boolean;
  className: string;
}) => (
  <span className={className}>
    <LinkWithRipple to={to} style={{ width: '100%', height: '100%' }}>
      <NavigationItem menuType={menuType} active={active} renderIcon={renderIcon} />
    </LinkWithRipple>
  </span>
);

const Navigation: React.FC<Props> = ({ currentMenuType, className = '' }) => {
  const classes = useStyles();
  return (
    <nav className={clsx(classes.root, className)}>
      <WrappedItem
        to={paths.training()}
        menuType={'training'}
        active={currentMenuType === 'training'}
        className={classes.itemWrapper}
      />
      <WrappedItem
        to={paths.exam()}
        menuType={'exam'}
        active={currentMenuType === 'exam'}
        className={classes.itemWrapper}
      />
      <WrappedItem
        to={paths.karutas()}
        menuType={'material'}
        active={currentMenuType === 'material'}
        className={classes.itemWrapper}
      />
      <WrappedItem
        to={paths.about()}
        menuType={'other'}
        active={currentMenuType === 'other'}
        className={classes.itemWrapper}
      />
    </nav>
  );
};

export default Navigation;
