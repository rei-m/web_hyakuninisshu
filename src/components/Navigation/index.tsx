import * as React from 'react';
import styled from 'styled-components';
import { withAppTheme } from '../../styles';
import { Link, LinkProps } from 'react-router-dom';
import NavIcon from '../NavIcon';
import { withRipple } from '../../enhancers/withRipple';
import { MenuType } from '../../enums';
import { ROUTE_PATHS } from '../../constants';

export interface NavigationProps {
  readonly currentMenuType?: MenuType;
}

const Root = withAppTheme(styled.nav)`
  display: flex;
  position: fixed;
  height: ${({ theme }) => theme.bottomNavHeight};
  width: 100vw;
  bottom: 0;
  background-color: ${({ theme }) => theme.colorPrimary};
  box-shadow: 0 -1px 5px rgba(0, 0, 0, 0.26);
  box-sizing: border-box;

  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    height: 100vh;
    width: ${({ theme }) => theme.sideNavWidth};
    padding-top: ${({ theme }) => theme.headerHeightWide};
    flex-direction: column;
  }
`;

const IconBox = styled.span`
  flex-grow: 1;
  width: 100%;
  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    flex-grow: 0;
    height: ${({ theme }) => theme.sideNavWidth};
  }
`;

const LinkWithRipple = withRipple<LinkProps>(Link);

const Navigation: React.SFC<NavigationProps> = ({ currentMenuType }) => (
  <Root>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.TRAINING}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Training}
          text="練習"
          isCurrent={currentMenuType === MenuType.Training}
        />
      </LinkWithRipple>
    </IconBox>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.EXAM}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Exam}
          text="腕試し"
          isCurrent={currentMenuType === MenuType.Exam}
        />
      </LinkWithRipple>
    </IconBox>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.KARUTAS}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Material}
          text="資料"
          isCurrent={currentMenuType === MenuType.Material}
        />
      </LinkWithRipple>
    </IconBox>
    <IconBox>
      <LinkWithRipple
        to={ROUTE_PATHS.ABOUT}
        style={{ width: '100%', height: '100%' }}
      >
        <NavIcon
          iconType={MenuType.Other}
          text="その他"
          isCurrent={currentMenuType === MenuType.Other}
        />
      </LinkWithRipple>
    </IconBox>
  </Root>
);

export default Navigation;
