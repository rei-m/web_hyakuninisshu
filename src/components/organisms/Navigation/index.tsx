import * as React from 'react';
import { GatsbyLinkProps, Link } from 'gatsby';
import styled from '@src/styles/styled-components';
import NavigationItem from '@src/components/molecules/NavigationItem';
import { menuIcon } from '@src/components/atoms/MenuIcon';
import { MenuType } from '@src/enums';
import { ROUTE_PATHS } from '@src/constants';
import { withRipple } from '@src/enhancers/withRipple';

export interface Props {
  currentMenuType?: MenuType;
  className?: string;
}

const Container = styled.nav`
  display: flex;
  background-color: ${({ theme }) => theme.colorPrimary};
  box-shadow: ${({ theme }) => theme.elevationShadow1x};
  box-sizing: border-box;
`;

// withRippleの作りがイマイチなせいで必要。。。
const ItemWrapper = styled.span`
  flex-grow: 1;
  width: 100%;
  @media screen and (min-width: ${({ theme }) => theme.minWidthWide}) {
    flex-grow: 0;
    height: ${({ theme }) => theme.sideNavWidth};
  }
`;

const LinkWithRipple = withRipple<GatsbyLinkProps<{}>>(Link);

const renderIcon = (menuType: MenuType) => {
  const Icon = menuIcon(menuType);
  return <Icon />;
};

const WrappedItem = ({ to, menuType, active }: { to: string; menuType: MenuType; active: boolean }) => (
  <ItemWrapper>
    <LinkWithRipple to={to} style={{ width: '100%', height: '100%' }}>
      <NavigationItem menuType={menuType} active={active} renderIcon={renderIcon} />
    </LinkWithRipple>
  </ItemWrapper>
);

const Navigation: React.FC<Props> = ({ currentMenuType, className }) => (
  <Container className={className}>
    <WrappedItem
      to={ROUTE_PATHS.TRAINING}
      menuType={MenuType.Training}
      active={currentMenuType === MenuType.Training}
    />
    <WrappedItem to={ROUTE_PATHS.EXAM} menuType={MenuType.Exam} active={currentMenuType === MenuType.Exam} />
    <WrappedItem to={ROUTE_PATHS.KARUTAS} menuType={MenuType.Material} active={currentMenuType === MenuType.Material} />
    <WrappedItem to={ROUTE_PATHS.ABOUT} menuType={MenuType.Other} active={currentMenuType === MenuType.Other} />
  </Container>
);

export default Navigation;
