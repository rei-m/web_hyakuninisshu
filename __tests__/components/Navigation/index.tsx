import * as ReactTestRenderer from 'react-test-renderer';
import * as React from 'react';
import Navigation from '@src/components/Navigation';
import { MenuType } from '@src/enums';

describe('<Navigation />', () => {
  it('should render component', () => {
    const renderer = ReactTestRenderer.create(<Navigation />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when current is training', () => {
    const renderer = ReactTestRenderer.create(<Navigation currentMenuType={MenuType.Training} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when current is exam', () => {
    const renderer = ReactTestRenderer.create(<Navigation currentMenuType={MenuType.Exam} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when current is material', () => {
    const renderer = ReactTestRenderer.create(<Navigation currentMenuType={MenuType.Material} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });

  it('should render component when current is other', () => {
    const renderer = ReactTestRenderer.create(<Navigation currentMenuType={MenuType.Other} />);
    expect(renderer.toJSON()).toMatchSnapshot();
  });
});
