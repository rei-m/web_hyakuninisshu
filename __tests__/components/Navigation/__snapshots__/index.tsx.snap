// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Navigation /> should render component 1`] = `
<nav
  className="sc-htpNat dtgQls"
>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/training"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hBfgJv"
        >
          create
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/exam"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hUDbCC"
        >
          note
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/karutas"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa kwyxYz"
        >
          library_books
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/about"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa fmFEyH"
        >
          settings
        </i>
      </a>
    </div>
  </span>
</nav>
`;

exports[`<Navigation /> should render component when current is exam 1`] = `
<nav
  className="sc-htpNat dtgQls"
>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/training"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hBfgJv"
        >
          create
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/exam"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa cNnMzU"
        >
          note
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/karutas"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa kwyxYz"
        >
          library_books
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/about"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa fmFEyH"
        >
          settings
        </i>
      </a>
    </div>
  </span>
</nav>
`;

exports[`<Navigation /> should render component when current is material 1`] = `
<nav
  className="sc-htpNat dtgQls"
>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/training"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hBfgJv"
        >
          create
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/exam"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hUDbCC"
        >
          note
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/karutas"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa dxCDSs"
        >
          library_books
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/about"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa fmFEyH"
        >
          settings
        </i>
      </a>
    </div>
  </span>
</nav>
`;

exports[`<Navigation /> should render component when current is other 1`] = `
<nav
  className="sc-htpNat dtgQls"
>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/training"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hBfgJv"
        >
          create
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/exam"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hUDbCC"
        >
          note
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/karutas"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa kwyxYz"
        >
          library_books
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/about"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa forfEu"
        >
          settings
        </i>
      </a>
    </div>
  </span>
</nav>
`;

exports[`<Navigation /> should render component when current is training 1`] = `
<nav
  className="sc-htpNat dtgQls"
>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/training"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa eZMEOL"
        >
          create
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/exam"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa hUDbCC"
        >
          note
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/karutas"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa kwyxYz"
        >
          library_books
        </i>
      </a>
    </div>
  </span>
  <span
    className="sc-bxivhb geoeNk"
  >
    <div
      className="sc-bwzfXH jRrEes"
      onMouseDown={[Function]}
      onMouseUp={[Function]}
    >
      <a
        color="#fff"
        href="/about"
        style={
          Object {
            "height": "100%",
            "width": "100%",
          }
        }
      >
        <i
          className="material-icons sc-bdVaJa fmFEyH"
        >
          settings
        </i>
      </a>
    </div>
  </span>
</nav>
`;
